diff -urN nbd-2.9.11.orig/gznbd/gznbd.c nbd-2.9.11/gznbd/gznbd.c
--- nbd-2.9.11.orig/gznbd/gznbd.c	2008-03-14 14:19:32 +0200
+++ nbd-2.9.11/gznbd/gznbd.c	2008-06-10 18:01:45 +0300
@@ -95,7 +95,7 @@
       fprintf(stderr,"%s: %s does not appear to be a valid size\n",argv[0],argv[3]);
       exit(1);
     }
-    printf("%s: file=%s, size=%Ld\n",argv[0],argv[2],size);
+    printf("%s: file=%s, size=%Ld\n",argv[0],argv[2],(long long)size);
   } else {
     char buffer[BLOCK];
     int result;
@@ -110,7 +110,7 @@
     }
 
     if(result==0){
-      printf("size=%Ld\n",size);
+      printf("size=%Ld\n",(long long)size);
     } else {
       printf("failed\n");
       if(result<0){
@@ -195,7 +195,7 @@
     from=ntohll(request.from);
 
 #ifdef TRACE
-fprintf(stderr,"%s: len=%d, from=%Ld\n",argv[0],len,from);
+fprintf(stderr,"%s: len=%d, from=%Ld\n",argv[0],len,(long long)from);
 #endif
 
     if(request.magic!=htonl(NBD_REQUEST_MAGIC)){
